Wbooks/Library/LibraryViewModel.swift

```diff
@@ -8,25 +8,12 @@
 import Foundation
 import UIKit
 
-struct LibraryViewModel {
+class LibraryViewModel {
     
     // MARK: Properties
-    private let booksArray : [Book] = [
-        Book(cover: UIImage(named: "BookCover1")!,
-             title: "A Little Bird Told Me",
-             author: "Timothy Cross"),
-        Book(cover: UIImage(named: "BookCover2")!,
-             title: "When the Doves Disappeared",
-             author: "Sofi Oksanen"),
-        Book(cover: UIImage(named: "BookCover3")!,
-             title: "The Best Book in the World",
-             author: "Peter Sjernstrom"),
-        Book(cover: UIImage(named: "BookCover4")!,
-             title: "Be Creative",
-             author: "Tony Alcazar"),
-        Book(cover: UIImage(named: "BookCover5")!,
-             title: "Redesign the Web",
-             author: "Liliana Castilla")]
+    let repository =  BookRepository()
```

Comment:
Let's use an abstraction instead of a concrete class. Here's what we'll do:

1. Add a protocol in BookRepository.swift:
```
protocol BookRepositoryType {
   func fetchBooks(...)
}
```
2. Make `BookRepository` conform to `BookRepositoryType`:
```
struct BookRepository: BookRepositoryType {
    ...
}
```

3. Change the type of this property to:
```
private let repository: BookRepositoryType
```
4. Then set it in the initializer:
```
init(repository: BookRepositoryType = BookRepository()) {
   self.repository = repository
}
```

We use abstractions for repositories because they have to be mocked later in tests, so this facilitates all the process 😉